crates/astria-composer/src/metrics_init.rs

```diff
@@ -0,0 +1,119 @@
+//! Crate-specific metrics functionality.
+//!
+//! Registers metrics & lists constants to be used as metric names throughout crate.
+
+use metrics::{
+    describe_counter,
+    describe_gauge,
+    describe_histogram,
+    Unit,
+};
+
+/// Labels
+pub(crate) const ROLLUP_ID_LABEL: &str = "rollup_id";
+pub(crate) const COLLECTOR_TYPE_LABEL: &str = "collector_type";
+
+/// Registers all metrics used by this crate.
+#[allow(clippy::too_many_lines)]
+pub fn register() {
+    describe_counter!(
+        TRANSACTIONS_RECEIVED,
+        Unit::Count,
+        "The number of transactions successfully received from collectors and bundled labelled by \
+         rollup"
+    );
+    describe_counter!(
+        TRANSACTIONS_DROPPED,
+        Unit::Count,
+        "The number of transactions dropped by the collectors before bundling it labelled by \
+         rollup and collector type"
+    );
+    describe_counter!(
+        TRANSACTIONS_DROPPED_TOO_LARGE,
+        Unit::Count,
+        "The number of transactions dropped because they were too large"
+    );
+    describe_counter!(
+        NONCE_FETCH_COUNT,
+        Unit::Count,
+        "The number of times we have attempted to fetch the nonce"
+    );
+    describe_counter!(
+        NONCE_FETCH_FAILURE_COUNT,
+        Unit::Count,
+        "The number of times we have failed to fetch the nonce"
+    );
+    describe_histogram!(
+        NONCE_FETCH_LATENCY,
+        Unit::Milliseconds,
+        "The latency of nonce fetch"
+    );
+    describe_gauge!(CURRENT_NONCE, Unit::Count, "The current nonce");
+    describe_histogram!(
+        SEQUENCER_SUBMISSION_LATENCY,
+        Unit::Milliseconds,
+        "The latency of submitting a transaction to the sequencer"
+    );
+    describe_counter!(
+        SEQUENCER_SUBMISSION_FAILURE_COUNT,
+        Unit::Count,
+        "The number of failed transaction submissions to the sequencer"
+    );
+    describe_histogram!(
+        TRANSACTIONS_PER_SUBMISSION,
+        Unit::Count,
+        "The number of rollup transactions successfully sent to the sequencer in a single \
+         submission"
+    );
+    describe_histogram!(
+        BYTES_PER_SUBMISSION,
+        Unit::Bytes,
+        "The total bytes successfully sent to the sequencer in a single submission"
+    );
+    describe_histogram!(
+        TOTAL_TRANSACTIONS_SUBMITTED,
+        Unit::Count,
+        "The total number of rollup transactions sent to the sequencer over the composer's \
+         lifetime. These can include failed submissions also"
+    );
+    describe_histogram!(
+        TOTAL_BYTES_SUBMITTED,
+        Unit::Bytes,
+        "The total bytes sent to the sequencer over the composer's lifetime. These can include \
+         failed submissions also"
+    );
```

Comment:
Note: These include failed submissions too. It ll be useful to know the difference b/w how many transactions and bytes were successfully submitted vs those which were not.